workflows/stairway.py

```diff
@@ -57,21 +57,36 @@ def ts_to_stairway(self, ts_path, num_bootstraps=1, mask_file=None):
             total_length += ts.sequence_length
             num_samples = ts.num_samples
             haps = ts.genotype_matrix()
-            # print(haps.shape)
+
+            # Mapping mutation type IDs to class of mutation (e.g., neutral, non-neutral)
+            mut_types = {}
+            for dfe in ts.metadata["stdpopsim"]["DFEs"]:
+                for mt in dfe["mutation_types"]:
+                    mid = mt["slim_mutation_type_id"]
+                    if not mid in mut_types:
+                        mut_types[mid] = "neutral" if mt["is_neutral"] else "non_neutral"
+
+            site_class = np.empty(ts.num_sites, dtype=object)
+
+            # Finding neutral positions
+            neu_positions = []
+            for j, s in enumerate(ts.sites()):
+                mt = []
+                for m in s.mutations:
+                    for md in m.metadata["mutation_list"]:
+                        mt.append(md["mutation_type"])
+                        if mut_types[md["mutation_type"]] == "neutral":
+                            neu_positions.append(m.site)
+                site_class[j] = mut_types[mt[0]] if len(mt) == 1 else "double_hit" 
+            assert sum(site_class == None) == 0
 
             SFSs = []
-            # Masking
-            # print("This is the number of mutations")
-            # print(ts.get_num_mutations())
-            # print("This is the number of variants")
-            # print(ts.variants())
-            # print("This is the number of sites")
-            # print(ts.num_sites)
-            #retain = np.full(ts.get_num_mutations(), False)
 
+            # Make it work, get ride off neutral positions that overlap the mask region
+            retain = np.full(ts.get_num_mutations(), False)
```

Comment:
The `ts.get_XXX()` methods have been depricated:
```suggestion
            retain = np.full(ts.num_mutations, False)
```